### ISSUE TITLE
ImageStack.write is inconsistent

### ISSUE THREAD

**Issue by ambrosejcarr** [spacetx/starfish#458]
When data is 3d, it writes 3d volumes in npy (1 per channel)
When data is 2d, it writes 2d tiffs in npy (1 per round, channel)

Both load fine when read in with `Experiment.from_json()`

It is possible that the problem is not this general, but try saving data from allen_smFISH. You get this: 
```
-rw-r--r--  1 ambrosecarr  staff     359 Aug 21 10:35 codebook.json
-rw-r--r--  1 ambrosecarr  staff     161 Aug 21 10:41 experiment.json
-rw-r--r--  1 ambrosecarr  staff  198128 Aug 21 10:33 hybridization-X0-Y0-Z0-H0-C0.npy
-rw-r--r--  1 ambrosecarr  staff  198128 Aug 21 10:33 hybridization-X0-Y0-Z0-H0-C1.npy
-rw-r--r--  1 ambrosecarr  staff  198128 Aug 21 10:33 hybridization-X0-Y0-Z0-H0-C2.npy
-rw-r--r--  1 ambrosecarr  staff   73696 Aug 21 10:33 hybridization.json
-rw-r--r--  1 ambrosecarr  staff  198128 Aug 21 10:34 nuclei-X0-Y0-Z0-H0-C0.npy
-rw-r--r--  1 ambrosecarr  staff     997 Aug 21 10:34 nuclei.json
```

**Comment by ambrosejcarr** [comment]
This method also apparently writes bad checksums in the allen_smFISH case.

**Comment by ambrosejcarr** [comment]
Fixed by 2334d2f0a2a91fd058d90597fa34b24e8740c074
